src/styles/main.scss

```
@@ -1,3 +1,81 @@
+@use "sass:math";
+
+$day-width: 100px;
+$gap: 1px;
+$week-column: 7;
+$gap-amount: 6;
+$calendar-padding: 10px;
+$calendar-width: (($day-width + $gap) * $week-column) - $gap;
+$days: 31;
+$font-size: 30px;
+
+@mixin square ($size) {
+  height: $size;
+  width: $size;
+}
+
+html {
+  font-family: Arial, sans-serif;
+  font-size: $font-size;
+}
+
 body {
   margin: 0;
 }
+
+* {
+  box-sizing: border-box;
+}
+
+.calendar {
+  display: flex;
+  margin: auto;
+  padding: $calendar-padding;
+  max-width: $calendar-width + (2 * $calendar-padding);
+  flex-wrap: wrap;
+  gap: $gap;
+
+  &__day {
+    display: flex;
+    @include square($day-width);
+    justify-content: center;
+    align-items: center;
+    border: 1px solid #000;
+    background-color: #eee;
+
+    @for $index from 1 through $days {
+      &--#{$index}::before {
+        content: "#{$index}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+      // animation-duration: 500ms;
+      transition: marging-top 0.5s;
```

В данном случае стоит использовать transition, т.к. анимация должна происходить после действия пользователя(hover). animation стоит использовать когда что-то должно меняться без участия пользователя

---

```suggestion
      // animation-duration: 500ms;
      transition: transform 0.5s;
```
Ты меняешь transform, а не margin, значит и transition должен применяться к transform(сейчас никакой анимации нет)

---

Так же, transition нужно прописывать не на hover, а на дефолтное состояние элемента, потому что в таком случае анимация будет работать только в одну сторону